src/app/sidebar/Sidebar.tsx

```diff
@@ -39,24 +39,31 @@ function computeParsedCourses(courses: Course[]) {
   );
 }
 
-function scrollToTop() {
-  const sideBarScroller = document.querySelector('#sideBarScroller');
-  if (sideBarScroller) sideBarScroller.scrollTop = 0;
-}
-
 export function Sidebar({ pid, setPid, subjects, courses }: SidebarProps): JSX.Element {
-  const parsedCourses = useMemo(() => computeParsedCourses(courses), [courses]);
   const [selectedSubject, setSelectedSubject] = useState<string | undefined>();
 
+  const parsedCourses = useMemo(() => computeParsedCourses(courses), [courses]);
+
   const handleSubjectChange = useCallback(
     (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
       const subject = e.currentTarget.getAttribute('data-subject');
       setSelectedSubject(subject ?? undefined);
-      scrollToTop();
     },
     [setSelectedSubject]
   );
 
+  const sortedSubjectCards = useMemo(
+    () =>
+      subjects
+        .sort((a, b) => (a.subject > b.subject ? 1 : b.subject > a.subject ? -1 : 0))
+        .map((subject, index) => (
+          <Box data-subject={subject.subject} onClick={handleSubjectChange} key={index}>
+            <Card subject={subject.subject} title={subject.title} />
+          </Box>
+        )),
+    [subjects]
```

Comment:
I did some reading into the `useMemo` and JSX elements. I think I would prefer we don't memoize the `JSX` elements itself as the expensive computation is the sorting. I believe what happens is if we do it with the elements we end up using more memory and I don't think storing JSX in memory is very beneficial and prefer not doing it.